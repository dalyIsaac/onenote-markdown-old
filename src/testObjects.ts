/**
 * This module contains objects which are to be used in the Jest tests
 */

import {
  Notebook as INotebook,
  OnenotePage as IPage,
  OnenoteSection as ISection,
  SectionGroup as ISectionGroup
} from "@microsoft/microsoft-graph-types";
import { User } from "msal";
import { Notebook } from "./types/Notebook";
import { Page } from "./types/Page";
import { Section } from "./types/Section";
import { SectionGroup } from "./types/SectionGroup";
import { UserData } from "./types/UserData";

export const userId = "jane.doe@email.com";
export const userId1 = "john.smith@email.com";

export const graphNotebookInstance: INotebook = {
  id: "notebookid",
  isDefault: true,
  isShared: false,
  lastModifiedDateTime: "datetime",
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl",
  sectionsUrl: "https://www.example.com/sections",
  userRole: "Owner"
};

export const graphNotebookInstance1: INotebook = {
  id: "notebookid1",
  isDefault: false,
  isShared: true,
  lastModifiedDateTime: "datetime1",
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl1",
  sectionsUrl: "https://www.example.com/sections1",
  userRole: "Owner"
};

export const graphNotebookInstance2: INotebook = {
  id: "notebookid2",
  isDefault: false,
  isShared: false,
  lastModifiedDateTime: "datetime2",
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl2",
  sectionsUrl: "https://www.example.com/sections2",
  userRole: "Reader"
};

export const notebook = new Notebook(
  graphNotebookInstance,
  userId,
  "genericstring"
);
export const notebook1 = new Notebook(
  graphNotebookInstance1,
  userId1,
  "genericstring1"
);
export const notebook2 = new Notebook(
  graphNotebookInstance2,
  userId,
  "genericstring"
);

export const graphSectionGroupInstance: ISectionGroup = {
  id: "sectionGroup",
  parentNotebook: graphNotebookInstance,
  sectionGroups: [],
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl",
  sections: [],
  sectionsUrl: "https://www.example.com/sections"
};

export const graphSectionGroupInstance1: ISectionGroup = {
  id: "sectionGroup1",
  parentNotebook: graphNotebookInstance,
  parentSectionGroup: graphSectionGroupInstance,
  sectionGroups: [],
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl1",
  sections: [],
  sectionsUrl: "https://www.example.com/sections"
};

export const sectionGroup = new SectionGroup(graphSectionGroupInstance, userId);
export const sectionGroup1 = new SectionGroup(
  graphSectionGroupInstance1,
  userId1
);

export const content = "<h1>Hello Page 1</h1>";
export const content1 = "<h1>Hello Page 2</h1>";

export const graphPageInstance: IPage = {
  content,
  contentUrl: "https://www.example.com/Hello%20Page%201/content",
  createdByAppId: "genericstring1",
  id: "page",
  lastModifiedDateTime: "2014-01-01T00:00:00Z",
  level: 0,
  order: 0,
  parentNotebook: graphNotebookInstance,
  title: "Hello Page 1",
  userTags: ["tag1", "tag2"]
};

export const graphSectionInstance: ISection = {
  id: "section",
  isDefault: true,
  pages: [graphPageInstance],
  pagesUrl: "https://www.example.com/pages",
  parentNotebook: graphNotebookInstance,
  parentSectionGroup: graphSectionGroupInstance
};

export const graphPageInstance1: IPage = {
  content: content1,
  contentUrl: "https://www.example.com/Hello%20Page%202/content",
  createdByAppId: "genericstring2",
  id: "page1",
  lastModifiedDateTime: "2014-01-01T00:00:00Z",
  level: 0,
  order: 0,
  parentNotebook: graphNotebookInstance,
  parentSection: graphSectionInstance,
  title: "Hello Page 2",
  userTags: ["tag1", "tag2"]
};

export const graphSectionInstance1: ISection = {
  id: "graphSection",
  isDefault: false,
  pages: [graphPageInstance1],
  pagesUrl: "https://www.example.com/pages1",
  parentNotebook: graphNotebookInstance1,
  parentSectionGroup: graphSectionGroupInstance1
};

export const section = new Section(
  graphSectionInstance,
  [[graphPageInstance, graphPageInstance1]],
  userId
);

export const section1 = new Section(
  graphSectionInstance1,
  [[graphPageInstance1]],
  userId1
);

export const page = new Page(graphPageInstance, userId);
export const page1 = new Page(graphPageInstance1, userId);

export const pageId = "genericstring1";

export const parentSelfUser = "jane.doe@email.com";

export const user = new UserData(
  new User(
    "john.smith@email.com",
    "John Smith",
    "identityProvider1",
    "genericstring1",
    {}
  ),
  "photoString1"
);
export const user1 = new UserData(
  new User(
    "jane.doe@email.com",
    "Jane Doe",
    "identityProvider2",
    "genericstring2",
    {}
  ),
  "photoString2"
);
