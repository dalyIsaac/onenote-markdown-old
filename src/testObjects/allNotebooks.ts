import { IStateUserNotebooks } from "../reducers";
import {
  graphNotebookInstance,
  graphNotebookInstance1,
  graphNotebookInstance2
} from "./graphNotebookInstances";
import { userId, userId1 } from "./users";

export const allNotebooks: IStateUserNotebooks[] = [
  {
    displayableId: "genericstring",
    notebooks: [graphNotebookInstance, graphNotebookInstance2],
    userId
  },
  {
    displayableId: "genericstring1",
    notebooks: [graphNotebookInstance1],
    userId: userId1
  }
];
