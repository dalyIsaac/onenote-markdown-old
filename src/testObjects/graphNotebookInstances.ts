import { Notebook as INotebook } from "@microsoft/microsoft-graph-types";

export const graphNotebookInstance: INotebook = {
  displayName: "notebook",
  id: "notebookid",
  isDefault: true,
  isShared: false,
  lastModifiedDateTime: "datetime",
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl",
  sectionsUrl: "https://www.example.com/sections",
  userRole: "Owner"
};

export const graphNotebookInstance1: INotebook = {
  displayName: "notebookid1",
  id: "notebookid1",
  isDefault: false,
  isShared: true,
  lastModifiedDateTime: "datetime1",
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl1",
  sectionsUrl: "https://www.example.com/sections1",
  userRole: "Owner"
};

export const graphNotebookInstance2: INotebook = {
  displayName: "notebookid2",
  id: "notebookid2",
  isDefault: false,
  isShared: false,
  lastModifiedDateTime: "datetime2",
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl2",
  sectionsUrl: "https://www.example.com/sections2",
  userRole: "Reader"
};
