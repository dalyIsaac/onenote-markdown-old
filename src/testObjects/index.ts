/**
 * This module contains objects which are to be used in the Jest tests
 */

export * from "./users";
export * from "./graphNotebookInstances";
export * from "./notebooks";
export * from "./graphSectionGroupInstances";
export * from "./sectionGroups";
import {
  OnenotePage as IPage,
  OnenoteSection as ISection
} from "@microsoft/microsoft-graph-types";
import { PersonaInitialsColor } from "office-ui-fabric-react";
import { IPersona } from "../components/users";
import { IStateOneNote, IStateUserNotebooks } from "../reducers";
import { Page } from "../types/Page";
import { Section } from "../types/Section";
import {
  graphNotebookInstance,
  graphNotebookInstance1,
  graphNotebookInstance2
} from "./graphNotebookInstances";
import {
  graphSectionGroupInstance,
  graphSectionGroupInstance1
} from "./graphSectionGroupInstances";
import { notebook, notebook1 } from "./notebooks";
import { sectionGroup, sectionGroup1 } from "./sectionGroups";
import { userId, userId1 } from "./users";

export const notebookOrder = ["notebookid", "notebookid1"];

export const content = "<h1>Hello Page 1</h1>";
export const content1 = "<h1>Hello Page 2</h1>";

export const graphPageInstance: IPage = {
  content,
  contentUrl: "https://www.example.com/Hello%20Page%201/content",
  createdByAppId: "genericstring1",
  id: "page",
  lastModifiedDateTime: "2014-01-01T00:00:00Z",
  level: 0,
  order: 0,
  parentNotebook: graphNotebookInstance,
  title: "Hello Page 1",
  userTags: ["tag1", "tag2"]
};

export const graphSectionInstance: ISection = {
  displayName: "section",
  id: "section",
  isDefault: true,
  pages: [graphPageInstance],
  pagesUrl: "https://www.example.com/pages",
  parentNotebook: graphNotebookInstance,
  parentSectionGroup: graphSectionGroupInstance
};

export const graphPageInstance1: IPage = {
  content: content1,
  contentUrl: "https://www.example.com/Hello%20Page%202/content",
  createdByAppId: "genericstring2",
  id: "page1",
  lastModifiedDateTime: "2014-01-01T00:00:00Z",
  level: 0,
  order: 0,
  parentNotebook: graphNotebookInstance,
  parentSection: graphSectionInstance,
  title: "Hello Page 2",
  userTags: ["tag1", "tag2"]
};

export const graphSectionInstance1: ISection = {
  id: "graphSection",
  isDefault: false,
  pages: [graphPageInstance1],
  pagesUrl: "https://www.example.com/pages1",
  parentNotebook: graphNotebookInstance1,
  parentSectionGroup: graphSectionGroupInstance1
};

export const section = new Section(
  graphSectionInstance,
  [[graphPageInstance, graphPageInstance1]],
  userId
);

export const section1 = new Section(
  graphSectionInstance1,
  [[graphPageInstance1]],
  userId1
);

export const page = new Page(
  graphPageInstance,
  userId,
  undefined,
  undefined,
  true
);
export const page1 = new Page(graphPageInstance1, userId);

export const pageId = "genericstring1";

export const parentSelfUser = "jane.doe@email.com";

export const allNotebooks: IStateUserNotebooks[] = [
  {
    displayableId: "genericstring",
    notebooks: [graphNotebookInstance, graphNotebookInstance2],
    userId
  },
  {
    displayableId: "genericstring1",
    notebooks: [graphNotebookInstance1],
    userId: userId1
  }
];

export const openedNotebooks: IStateOneNote = {};
openedNotebooks[notebook.id] = notebook;

export const persona: IPersona = {
  imageInitials: "JS",
  imageUrl:
    "https://upload.wikimedia.org/wikipedia/commons/4/44/Microsoft_logo.svg",
  initialsColor: PersonaInitialsColor.blue,
  personaName: "john.smith@outlook.com",
  userName: "John Smith"
};

export const persona1: IPersona = {
  imageInitials: "JD",
  imageUrl:
    "https://upload.wikimedia.org/wikipedia/commons/9/9e/Microsoft_OneNote_2013_logo.svg",
  initialsColor: PersonaInitialsColor.green,
  personaName: "jane.doe@outlook.com",
  userName: "Jane Doe"
};

export const onenote: IStateOneNote = {};
onenote[notebook.id] = notebook;
onenote[notebook1.id] = notebook1;
onenote[sectionGroup.id] = sectionGroup;
onenote[sectionGroup1.id] = sectionGroup1;
onenote[section.id] = section;
onenote[section1.id] = section1;
onenote[page.id] = page;
onenote[page1.id] = page1;

export const selectedNav: string[] = [
  notebook1.id,
  sectionGroup1.id,
  section1.id,
  page1.id
];
