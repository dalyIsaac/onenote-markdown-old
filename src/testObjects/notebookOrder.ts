export const notebookOrder = ["notebookid", "notebookid1"];
