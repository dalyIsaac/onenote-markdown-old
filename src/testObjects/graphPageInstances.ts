import { OnenotePage as IPage } from "@microsoft/microsoft-graph-types";
import { graphNotebookInstance } from "./graphNotebookInstances";
import { graphSectionInstance } from "./index";

export const content = "<h1>Hello Page 1</h1>";
export const content1 = "<h1>Hello Page 2</h1>";

export const graphPageInstance: IPage = {
  content,
  contentUrl: "https://www.example.com/Hello%20Page%201/content",
  createdByAppId: "genericstring1",
  id: "page",
  lastModifiedDateTime: "2014-01-01T00:00:00Z",
  level: 0,
  order: 0,
  parentNotebook: graphNotebookInstance,
  title: "Hello Page 1",
  userTags: ["tag1", "tag2"]
};

export const graphPageInstance1: IPage = {
  content: content1,
  contentUrl: "https://www.example.com/Hello%20Page%202/content",
  createdByAppId: "genericstring2",
  id: "page1",
  lastModifiedDateTime: "2014-01-01T00:00:00Z",
  level: 0,
  order: 0,
  parentNotebook: graphNotebookInstance,
  parentSection: graphSectionInstance,
  title: "Hello Page 2",
  userTags: ["tag1", "tag2"]
};
