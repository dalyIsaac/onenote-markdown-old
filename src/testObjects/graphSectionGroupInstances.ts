import { SectionGroup as ISectionGroup } from "@microsoft/microsoft-graph-types";
import { graphNotebookInstance } from "./graphNotebookInstances";

export const graphSectionGroupInstance: ISectionGroup = {
  displayName: "sectionGroup",
  id: "sectionGroup",
  parentNotebook: graphNotebookInstance,
  sectionGroups: [],
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl",
  sections: [],
  sectionsUrl: "https://www.example.com/sections"
};

export const graphSectionGroupInstance1: ISectionGroup = {
  id: "sectionGroup1",
  parentNotebook: graphNotebookInstance,
  parentSectionGroup: graphSectionGroupInstance,
  sectionGroups: [],
  sectionGroupsUrl: "https://www.example.com/sectionGroupsUrl1",
  sections: [],
  sectionsUrl: "https://www.example.com/sections"
};
