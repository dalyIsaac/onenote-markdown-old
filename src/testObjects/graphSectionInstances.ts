import { OnenoteSection as ISection } from "@microsoft/microsoft-graph-types";
import {
  graphNotebookInstance,
  graphNotebookInstance1
} from "./graphNotebookInstances";
import { graphPageInstance, graphPageInstance1 } from "./graphPageInstances";
import {
  graphSectionGroupInstance,
  graphSectionGroupInstance1
} from "./graphSectionGroupInstances";

export const graphSectionInstance: ISection = {
  displayName: "section",
  id: "section",
  isDefault: true,
  pages: [graphPageInstance],
  pagesUrl: "https://www.example.com/pages",
  parentNotebook: graphNotebookInstance,
  parentSectionGroup: graphSectionGroupInstance
};

export const graphSectionInstance1: ISection = {
  id: "graphSection",
  isDefault: false,
  pages: [graphPageInstance1],
  pagesUrl: "https://www.example.com/pages1",
  parentNotebook: graphNotebookInstance1,
  parentSectionGroup: graphSectionGroupInstance1
};
