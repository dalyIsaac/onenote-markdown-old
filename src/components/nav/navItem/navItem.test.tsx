import * as enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import { Callout, Icon, setIconOptions } from "office-ui-fabric-react";
import * as React from "react";
import * as renderer from "react-test-renderer";
import { sectionGroup } from "../../../testObjects";
import NavItem, { IPropsNavItem } from "./navItem";

setIconOptions({
  disableWarnings: true
});

enzyme.configure({ adapter: new Adapter() });

function setUp() {
  const props: IPropsNavItem = {
    icon: <Icon iconName="Sections" />,
    indentation: 0,
    isSelected: false,
    item: sectionGroup,
    key: "SectionGroupKey",
    navItemContexts: [
      <p key="p">Hello world</p>,
      <p key="p1">Hello world 1</p>
    ],
    updateIsExpanded: jest.fn(),
    updateSelected: jest.fn()
  };
  const wrapper = enzyme.mount(<NavItem {...props} />);
  return { props, wrapper };
}

describe("Components: NavItem", () => {
  test("It should render itself and subcomponents", () => {
    const { props, wrapper } = setUp();

    const parentDiv = wrapper.find("div").first();
    expect(parentDiv.props().style).toEqual({ marginLeft: 0 });

    const button = wrapper.find("button").first();
    expect(button.hasClass("navItem")).toBe(true);
    expect(button.parent().is("div")).toBe(true);

    const divNavItemWrapper = wrapper.find("div").at(1);
    expect(divNavItemWrapper.hasClass("navItemWrapper")).toBe(true);
    expect(divNavItemWrapper.parent().is("button.navItem")).toBe(true);

    const icon = wrapper.find(Icon);
    expect(icon.props().iconName).toBe("Sections");
    expect(icon.parent().is("div.navItemWrapper")).toBe(true);

    const label = wrapper.find("label");
    expect(label.text()).toBe(props.item.displayName);
    expect(label.parent().is("div.navItemWrapper")).toBe(true);
  });

  test("Right click on button, and the item is not already selected", () => {
    const { wrapper, props } = setUp();
    const button = wrapper.find("button").first();
    button.simulate("contextmenu");
    const instance = wrapper.instance() as NavItem;

    // item is not selected
    // section groups are expandable
    // the section group is not expanded
    // calls updateSelected
    const updateSelected = instance.props.updateSelected as jest.Mock;
    expect(updateSelected.mock.calls.length).toBe(1);
    expect(updateSelected.mock.calls[0]).toEqual([props.item.id]);

    // calls updateIsExpanded
    const updateIsExpanded = instance.props.updateIsExpanded as jest.Mock;
    expect(updateIsExpanded.mock.calls.length).toBe(1);
    expect(updateIsExpanded.mock.calls[0]).toEqual([props.item.id, true]);

    expect(instance.state.rightClick).toBe(true);

    // Callout should be rendered
    const callout = wrapper.find(Callout);
    expect(callout.parent().is("div"));
    expect(
      renderer.create((wrapper.instance() as NavItem).callout).toJSON()
    ).toMatchSnapshot();
  });

  test("Right click on button, and the item is already selected", () => {
    const { wrapper } = setUp();
    wrapper.setProps({ isSelected: true });
    const button = wrapper.find("button").first();
    button.simulate("contextmenu");
    const instance = wrapper.instance() as NavItem;

    // item is selected
    const updateSelected = instance.props.updateSelected as jest.Mock;
    expect(updateSelected.mock.calls.length).toBe(1);
    expect(updateSelected.mock.calls[0]).toEqual(["notebookid"]);

    // item is expandable
    const updateIsExpanded = instance.props.updateIsExpanded as jest.Mock;
    expect(updateIsExpanded.mock.calls.length).toBe(1);
    expect(updateIsExpanded.mock.calls[0]).toEqual(["sectionGroup", false]);

    expect(instance.state.rightClick).toBe(true);

    // Callout should be rendered
    const callout = wrapper.find(Callout);
    expect(callout.parent().is("div"));
    expect(
      renderer.create((wrapper.instance() as NavItem).callout).toJSON()
    ).toMatchSnapshot();
  });

  // displayName slice 40

  // page, because title

  // item which is selected
});
