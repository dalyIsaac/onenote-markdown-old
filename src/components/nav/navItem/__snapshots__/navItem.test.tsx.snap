// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: NavItem Page Right click on button, and the item is not already selected 1`] = `
<span
  className="ms-Layer"
/>
`;

exports[`Components: NavItem Section Right click on button, and the item is already selected 1`] = `
<span
  className="ms-Layer"
/>
`;

exports[`Components: NavItem Section Right click on button, and the item is not already selected 1`] = `
<span
  className="ms-Layer"
/>
`;

exports[`Components: NavItem Section group Right click on button, and the item is already selected 1`] = `
<span
  className="ms-Layer"
/>
`;

exports[`Components: NavItem Section group Right click on button, and the item is not already selected 1`] = `
<span
  className="ms-Layer"
/>
`;
